src/server/pps/server/api_server.go

```diff
@@ -341,7 +341,22 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 		}
 		outputCommit, err = pfsAPIClient.StartCommit(ctx, startCommitRequest)
 		if err != nil {
-			return nil, err
+			if strings.Contains(err.Error(), "already exists") {
+				if _, err := pfsAPIClient.DeleteCommit(
+					ctx,
+					&pfsclient.DeleteCommitRequest{
+						Commit: client.NewCommit(parent.Repo.Name, strings.Split(parent.ID, "/")[0]),
```

Comment:
I would also add some debug log statements just so we know if/when this code path has been hit.